include/lsst/afw/geom/Transform.h

```diff
@@ -0,0 +1,139 @@
+// -*- lsst-c++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_TRANSFORM_H
+#define LSST_AFW_GEOM_TRANSFORM_H
+
+#include <memory>
+#include <vector>
+
+#include "astshim.h"
+#include "ndarray.h"
+
+#include "lsst/afw/geom/Endpoint.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+/**
+Transform LSST spatial data, such as Point2D and SpherePoint, using an AST transform.
+
+This class contains two Endpoints, to specify the "from" and "to" LSST data type,
+and an astshim::FrameSet or astshim::Mapping to specify the transformation.
```

Comment:
A `FrameSet` may contain multiple transformations, but Transform only exposes the mapping that connects the `BASE` frame to the `CURRENT` frame. I added the following line to the comment in hopes of making this clearer:
```
In the case of a FrameSet the transformation is from the `BASE` frame to the `CURRENT` frame.
```